### PR TITLE
Add Envoy example for OTEL collector

### PR DESCRIPTION
**Pull request by nslaughter**
## What it is

This is the running example, which is based on the front-proxy
example from the github.com/envoyproxy/envoy/examples/front-proxy. 

### Build images first. 
Run these commands ...
  a. docker compose build flask_service
  b. docker compose build tracing

Note that the build for these images is specified in the docker compose file. They're the services with 0 replicas. And compose has a strange UI where when you combine the `image` and the `build` fields you get a build tagged with the value you provide for `image`. It's weird, but turns out to simplify things because the compose file can be a holistic configuration.

### Run

In addition to the commands here, you'll need to also mind the steps outlined in the README, namely you need a Lightstep access token stored in the proper env var.

`docker compose up` will bring up the environment.

`docker compose --profile loadgen up` also brings up a container to send traffic to front-proxy.

### CODE REVIEW

**Review by mwear — COMMENTED** [review]
One minor suggestion on the intro, but otherwise this looks good!

File: collector/envoy/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Ingest Envoy metrics using OTEL Collector's Prometheus receiver
+
+The OTEL Collector has a variety of [third party receivers](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/master/receiver) that provide integration with a wide variety of metric sources.
```
[the remark this answers is not in the document]
Comment (nslaughter) [reply collector/envoy/README.md]: Thanks for this @mwear. The approach makes sense.

I've addressed this with updated language as described, so I'm marking it resolved. However, if this missed the mark feel free to ping me again on it.

**Comment by mwear** [comment]
This looks good to me. If you are ready to bring this out of draft, it might encourage more reviews / feedback.

**Comment by nslaughter** [comment]
hi, @mwear and @carlosalberto. Pinging to see that we get this merged.

**Pull request merged**